### ISSUE TITLE
Analysis: Effect of external weather conditions on the power consumption within the container

### ISSUE THREAD

**Issue by Yapping72** [danialhbma/ITP-SE12-Power-Monitoring#30]
Research on possible methods to analyze how external weather conditions impact power consumption within the container.

**Comment by danialhbma** [comment]
Plan as of now:

- Use OpenWeather API data to observe weather trends (in conjunction with correlation algorithm to observe the relationship between external weather and container temperature fluctuations)
- Use NEA historical data to track seasonal weather trends to predict power consumption costs

**Comment by Yapping72** [comment]
Created InfluxDB read gateway that allows us to retrieve measurements in our buckets via Python. Bucket data can be loaded in pandas dataframe or as .csv filles. This should provide an easier way to perform analysis in Python. 

May also use this ReadGateWay to track growlight alerts if alerts via grafana still buggy. 
![Image](https://github.com/danialhbma/ITP-SE12-Power-Monitoring/assets/54227281/cb90ffff-0173-450d-991b-f2f0be734240)

**Comment by Yapping72** [comment]
@chococtea Sarah will help with creating the correlation analysis script to analyze: 
> Correlation between WeatherAPI bucket and predicted power consumption (from her model model).  

Will take over the plotting of text to heatmap for both our correlation analysis.

**Comment by Yapping72** [comment]
Use CorrelationCalculator provided by @chococtea  to study correlation between external farm conditions and internal farm conditions.
![WeatherAPI   Farm Conditions](https://github.com/danialhbma/ITP-SE12-Power-Monitoring/assets/54227281/0471a012-985a-4c98-80d2-9e6a68657d3a)

Following which, we removed weakly correlated values. We then analyzed how external and internal conditions that are strongly correalted can have effect on estimated farm power consumption.
![WeatherAPI   Farm Conditions   Farm Power Consumption](https://github.com/danialhbma/ITP-SE12-Power-Monitoring/assets/54227281/8875428f-8de8-45b6-aad4-e202d95aac9a)
